docs/src/design/proposals/APIRedesign_PeteRichards.md

```diff
@@ -0,0 +1,251 @@
+**Table of Contents**  *generated with [DocToc](https://github.com/thlorenz/doctoc)*
+
+- [Reducing interface depth (the bundle.json version)](#reducing-interface-depth-the-bundlejson-version)
+  - [Imperitive component registries](#imperitive-component-registries)
+  - [Get rid of "extension category" concept.](#get-rid-of-extension-category-concept)
+  - [Reduce number and depth of extension points](#reduce-number-and-depth-of-extension-points)
+  - [Composite services should not be the default](#composite-services-should-not-be-the-default)
+  - [Get rid of views, representations, and templates.](#get-rid-of-views-representations-and-templates)
+- [Reducing interface depth (The angular discussion)](#reducing-interface-depth-the-angular-discussion)
+  - [More angular: for all services](#more-angular-for-all-services)
+  - [Less angular: only for views](#less-angular-only-for-views)
+- [Standard packaging and build system](#standard-packaging-and-build-system)
+  - [Use systemjs for module loading](#use-systemjs-for-module-loading)
+  - [Use gulp or grunt for standard tooling](#use-gulp-or-grunt-for-standard-tooling)
+  - [Package openmctweb as single versioned file.](#package-openmctweb-as-single-versioned-file)
+- [Misc Improvements](#misc-improvements)
+  - [Refresh on navigation](#refresh-on-navigation)
+  - [Move persistence adapter to promise rejection.](#move-persistence-adapter-to-promise-rejection)
+  - [Remove bulk requests from providers](#remove-bulk-requests-from-providers)
+- [Notes on current API proposals:](#notes-on-current-api-proposals)
+- [[1] Footnote: The angular debacle](#1-footnote-the-angular-debacle)
+  - ["Do or do not, there is no try"](#do-or-do-not-there-is-no-try)
+  - [A lack of commitment](#a-lack-of-commitment)
+  - [Commitment is good!](#commitment-is-good)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+
+# Reducing interface depth (the bundle.json version)
+
+## Imperitive component registries 
+
+Transition component registries to javascript, get rid of bundle.json and bundles.json.  Prescribe a method for application configuration, but allow flexibility in how application configuration is defined.
+
+Register components in an imperitive fashion, see angularApp.factory, angularApp.controller, etc.  Alternatively, implement our own application object with new registries and it's own form of registering objects.
+
+## Get rid of "extension category" concept.
+
+The concept of an "extension category" is itself an extraneous concept-- an extra layer of interface depth, an extra thing to learn before you can say "hello world".  Extension points should be clearly supported and documented with whatever interfaces make sense.  Developers who wish to add something that is conceptually equivalent to an extension category can do so directly, in the manner that suites their needs, without us forcing a common method on them.
```

Comment:
This proposal is a "policy statement" about how we communicate these concepts to developers and users more so than a specific requirement for code.

Yes, Registering providers should have consistent syntax, but that's just good programming; it's not a feature.  

The developer guide should show how to extend the platform while introducing the bare minimum number of concepts. "RegistryProvider" is a concept that they just don't need to know about.